crates/nu_plugin_str/src/strutils.rs

```diff
@@ -282,6 +293,14 @@ pub mod tests {
         Ok(())
     }
 
+    #[test]
+    fn converts_to_float() -> Result<(), Box<dyn std::error::Error>> {
+        let mut strutils = Str::new();
+        strutils.for_to_float();
+        assert_eq!(strutils.apply("3.1415")?, decimal(3.1415).value);
```

Comment:
I think the problem maybe be that `impl Into<BigDecimal> for f32` could be the problem. What if you made a decimal implementation that took a `&str` and used the `FromStr` implementation for `BigDecimal`?

Reply:
This test is a pass, I made wrong link anchor by mistake. This is the one that fails because of trailing zeros -  [tests/plugins/core_str.rs](https://github.com/nushell/nushell/pull/1872/files#diff-f6135a125bc0a55b9b470fca0e4223e0). It looks like there will be solution to this problem in next version of bigdecimal https://github.com/akubera/bigdecimal-rs/pull/63, although I will a bit more digging to understand why they did appear in the first place